WordPress/gutenberg
# Interface package: move MainDashboardButton slot

## Description

See the related discussion here https://github.com/WordPress/gutenberg/commit/410333bd86df766631ef4879b613be9834e34912#commitcomment-44381621.

The purpose of this change is to safeguard against similar problems in the future. It might also make more sense to organize things this way now, since the buttons being replaced with the slot are no longer the same. In post editor it's the W icon button, while in site editor it's `Back to Dashboard` nav item. That means that in 3rd party integration we'd always have to check the editor type before providing the appropriate fill with the old approach.

Closes https://github.com/WordPress/gutenberg/issues/27158

## How has this been tested?




1. Make sure that default cases work with no fills work in both editors.
2. Try adding some custom buttons:

For post editor:

```js
import { registerPlugin } from '@wordpress/plugins';
import {
	__experimentalFullscreenModeClose as FullscreenModeClose,
	__experimentalMainDashboardButton as MainDashboardButton,
} from '@wordpress/edit-post';
import { close } from '@wordpress/icons';


const MainDashboardButtonIconTest = () => (
    <MainDashboardButton>
        <FullscreenModeClose icon={ close } href="http://wordpress.org" />
    </MainDashboardButton>
);

registerPlugin( 'main-dashboard-button-icon-test', {
	render: MainDashboardButtonIconTest,
} );
```

For site editor:

```js
import { registerPlugin } from '@wordpress/plugins';
import {
	__experimentalMainDashboardButton as MainDashboardButton,
} from '@wordpress/edit-site';
import {
	__experimentalNavigationBackButton as NavigationBackButton,
} from '@wordpress/components';

const MainDashboardButtonIconTest = () => (
    <MainDashboardButton>
        <NavigationBackButton
            backButtonLabel={ __( 'Back to the future' ) }
            className="edit-site-navigation-panel__back-to-dashboard"
            href="http://wordpress.org/"
        />
    </MainDashboardButton>
);

registerPlugin( 'main-dashboard-button-icon-test', {
	render: MainDashboardButtonIconTest,
} );
```

## Screenshots 

## Types of changes



Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
